### ISSUE TITLE
Domain dialog should use standard dialog patterns

### ISSUE THREAD

**Issue by stefwalter** [cockpit-project/cockpit#3415]
The 'Join Domain' dialog should use standard dialog patterns:

https://github.com/cockpit-project/cockpit/wiki/HIG:-Dialogs

Those patterns should be implemented using code like this:

 * https://github.com/cockpit-project/cockpit/blob/master/pkg/playground/patterns.js
 * https://github.com/cockpit-project/cockpit/blob/master/pkg/playground/patterns.html

**Comment by stefwalter** [comment]
Two examples of incorrect behavior here:

![screenshot from 2016-01-05 11-08-17](https://cloud.githubusercontent.com/assets/795070/12112409/ae907a14-b39c-11e5-9c01-c23b6daaf19a.png)

The failure message should be at the bottom of the dialog.

![screenshot from 2016-01-05 11-08-36](https://cloud.githubusercontent.com/assets/795070/12112418/bbac0772-b39c-11e5-91a6-05a5094b199d.png)

The spinner should be the standard spinner, no need for 'This may take a while' and also the previous error message has not been cleared.

**Comment by abhishekgupta5** [comment]
@stefwalter > Is this bug still open ? If yes, can you tell the source of the file that needs work ?

**Comment by andreasn** [comment]
Yes, the issue is still open. It was only issue #2830 that was closed with commit 3efcc930880907f58fd883b236c454aa63a22ee5

**Comment by andreasn** [comment]
abhishekgupta5: Marius invited you to the group. I think I'm able to assign you after that. Feel free to start working on it regardless though :)

**Comment by abhishekgupta5** [comment]
@stefwalter  @andreasn > Can you tell me the source of the file which needs change ?

**Comment by andreasn** [comment]
That would be one of these files I think https://github.com/cockpit-project/cockpit/tree/master/pkg/realmd

**Comment by abhishekgupta5** [comment]
@stefwalter  @andreasn  > Where exactly do you expect the failure message to appear ? Is it to the bottom left part of the dialogue (parallel to the 'Cancel' and 'Join' button) ? A screenshot with a highlighted area(where exactly should the message appear) can be a great help so as to get an idea of padding also

**Comment by stefwalter** [comment]
If you 'sudo make install' cockpit you'll see 'Design Patterns' under tools. You can see how a dialog should be have in that screen, including where the error messages go, and how to use the standard patterns.

**Comment by martinpitt** [comment]
It is *fairly* standard now, but indeed it should be ported to ct-form-layout.

**Comment by martinpitt** [comment]
Oh, nevermind, this also applies to the error message. Yes, this should also be standardized.
